pyriemann/transferlearning.py

```diff
@@ -0,0 +1,158 @@
+    ----------
+    transfer_coef : float
+        Transfer coefficient in [0,1], controlling the trade-off between
+        source and target data. At 0, there is no transfer, only the data
+        acquired from the source are used. At 1, this is a calibration-free
+        system as no data are required from the source.
+    metric : string | dict, default='riemann'
+        The type of metric used for centroid and distance estimation.
+        see `mean_covariance` for the list of supported metric.
+        the metric could be a dict with two keys, `mean` and `distance` in
+        order to pass different metric for the centroid estimation and the
+        distance estimation. Typical usecase is to pass 'logeuclid' metric for
+        the mean in order to boost the computional speed and 'riemann' for the
+        distance in order to keep the good sensitivity for the classification.
+    n_jobs : int, default=1
+        The number of jobs to use for the computation. This works by computing
+        each of the class centroid in parallel.
+        If -1 all CPUs are used. If 1 is given, no parallel computing code is
+        used at all, which is useful for debugging. For n_jobs below -1,
+        (n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs but one
+        are used.
+
+    Attributes
+    ----------
+    covmeans_ : list
+        Class centroids, estimated after fit.
+    classes_ : list
+        List of classes, obtained after fit
+
+    See Also
+    --------
+    MDM
+
+    References
+    ----------
+    .. [1] E. Kalunga, S. Chevallier and Q. Barthelemy, "Transfer learning for
+        SSVEP-based BCI using Riemannian similarities between users", in 26th
+        European Signal Processing Conference (EUSIPCO), pp. 1685-1689. IEEE,
+        2018.
+
+    .. [2] S. Khazem, S. Chevallier, Q. Barthelemy, K. Haroun and C. Nous,
+        "Minimizing Subject-dependent Calibration for BCI with Riemannian
+        Transfer Learning", in 10th International IEEE/EMBS Conference on
+        Neural Engineering (NER), pp. 523-526. IEEE, 2021.
+    """
+
+    def __init__(self, transfer_coef, metric='riemann', n_jobs=1):
+        """Init."""
+        self.transfer_coef = transfer_coef
+        self.metric = metric
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, X_source, y_source, sample_weight=None):
+        """Fit (estimates) the centroids.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices from target domain.
+        y : ndarray, shape (n_matrices,)
+            Labels for each matrix of target domain.
+        X_source : ndarray, shape (n_matrices_source, n_channels, n_channels)
+            Set of SPD matrices from source domain.
+        y_source : ndarray, shape (n_matrices_source,)
+            Labels for each matrix of source domain.
+        sample_weight : None | ndarray, shape (n_matrices_source,), default=None
```

Comment:
This modification will remove flake8 error.
```suggestion
        sample_weight : None | ndarray, shape (n_matrices_source,), \
            default=None
```